src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg payload: request containing (dtype+name)
+    :type payload: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc genAttachMsg(cmd: string, payload: string, 
+                                            st: borrowed SymTab): MsgTuple throws {
+        var repMsg: string; // response message
+
+        var ele_parts = payload.split("+");
+        var dtype = ele_parts[0];
+        var name = ele_parts[1];
+
+        if dtype == "infer" {
+            // Try to determine the type from the entries in the symbol table
+            // Should try __RegisteredSymbols__ instead
+            var symbols = st.info("__RegisteredSymbols__");
+            regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "Attempting to find type of registered element '%s'".format(name));
+
+            if st.contains(name) {
+                // Easy case where full name given matches an entry, pdarray or Strings
+                dtype = "simple";
+            } else if st.contains("%s.categories".format(name)) && st.contains("%s.codes".format(name)) {
+                dtype = "categorical";
+            } else if st.contains("%s_segments".format(name)) && st.contains("%s_values".format(name)) {
+                // Important to note that categorical has a .segments while segarray uses _segments
+                dtype = "segarray";
+            } else {
+                throw getErrorWithContext(
+                                    msg="Unable to determine type for given name: %s".format(name),
+                                    lineNumber=getLineNumber(),
+                                    routineName=getRoutineName(),
+                                    moduleName=getModuleName(),
+                                    errorClass="ValueError"
+                                    );
+            }
+        }
+
+        // type possibilities for pdarray and strings
+        var simpleTypes: list(string) = ["pdarray","int64", "uint8", "uint64", "float64", "bool", "strings", "string", "str"];
+        if simpleTypes.contains(dtype.toLower()) {
+            dtype = "simple";
+        }
+
+        select (dtype.toLower()) {
+            when ("simple") {
+                // pdarray and strings can use the attachMsg method
+                return attachMsg(cmd, name, st);
+            }
+            when ("categorical") {
+                return attachCategoricalMsg(cmd, name, st);
+            }
+            when ("segarray") {
+                return attachSegArrayMsg(cmd, name, st);
+            }
+            otherwise {
+                regLogger.warn(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "Unsupported type provided: '%s'. Supported types are: int64, str, categorical, segarray".format(dtype));
```

Comment:
should this log message say `pdarray` instead of `int64`?

Reply:
Yes it should